### PR TITLE
fix(agw): Modifed code to port hash list of ue contexts to protobuf map

### PR DESCRIPTION
**Pull request by rsarwad**
## Summary
fix(agw): Modifed code to port hash list of ue contexts to protobuf map
## Test Plan

Executed s1ap sanity test suite and unit test cases.

## Additional Information
This PR is duplicate of https://github.com/magma/magma/pull/13895 because of DCO check failure

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/sgw/sgw_context_manager.cpp [hunk 1]
```diff
@@ -116,21 +116,9 @@ mme_sgw_tunnel_t* sgw_cm_create_s11_tunnel(teid_t remote_teid,
...
-     * Malloc failed, may be ENOMEM error
-     */
-    OAILOG_ERROR(LOG_SPGW_APP,
-                 "Failed to create tunnel for remote_teid " TEID_FMT "\n",
-                 remote_teid);
-    return NULL;
-  }
+  mme_sgw_tunnel_t* new_tunnel = nullptr;
 
+  new_tunnel = new mme_sgw_tunnel_t();
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/sgw_context_manager.cpp]: Can be merged in one line.

File: lte/gateway/c/core/oai/test/spgw_task/test_spgw_state_converter.cpp [hunk 2]
```diff
@@ -76,8 +76,8 @@ TEST(SPGWStateConverterTest, TestUEContextConversion) {
     auto want_teid = LIST_FIRST(&initial_ctx->sgw_s11_teid_list)->sgw_s11_teid;
 
     oai::SpgwUeContext proto_ctx;
-    spgw_ue_context_t* final_ctx =
-        (spgw_ue_context_t*)calloc(1, sizeof(spgw_ue_context_t));
+    spgw_ue_context_t* final_ctx = new spgw_ue_context_t();
+    EXPECT_NE(final_ctx, nullptr);
```
Comment (ajahl) [inline lte/gateway/c/core/oai/test/spgw_task/test_spgw_state_converter.cpp]: Line can be removed. Is not needed.

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp [hunk 3]
```diff
@@ -137,7 +137,7 @@ spgw_ue_context_t* sgw_create_or_get_ue_context(sgw_state_t* sgw_state,
   hashtable_ts_get(sgw_state->imsi_ue_context_htbl, (const hash_key_t)imsi64,
                    (void**)&ue_context_p);
   if (!ue_context_p) {
-    ue_context_p = (spgw_ue_context_t*)calloc(1, sizeof(spgw_ue_context_t));
+    ue_context_p = new spgw_ue_context_t();
     if (ue_context_p) {
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp]: nullptr check does not work here. Remove it or use try..catch.

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp [hunk 4]
```diff
@@ -137,7 +137,7 @@ spgw_ue_context_t* sgw_create_or_get_ue_context(sgw_state_t* sgw_state,
   hashtable_ts_get(sgw_state->imsi_ue_context_htbl, (const hash_key_t)imsi64,
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp:137]: https://github.com/magma/magma/blob/2ae5901bc4efb6796b48c61c5f9071c4cbdafae9/lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp#L136
Can you replace NULL with nullptr  
```cpp
spgw_ue_context_t* ue_context_p = nullptr;
```

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp [hunk 5]
```diff
@@ -162,8 +162,7 @@ status_code_e sgw_update_teid_in_ue_context(sgw_state_t* sgw_state,
     OAILOG_FUNC_RETURN(LOG_SGW_S8, RETURNerror);
   }
 
-  sgw_s11_teid_t* sgw_s11_teid_p =
-      (sgw_s11_teid_t*)calloc(1, sizeof(sgw_s11_teid_t));
+  sgw_s11_teid_t* sgw_s11_teid_p = new sgw_s11_teid_t();
   if (!sgw_s11_teid_p) {
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp]: nullptr check does not work here. Remove it or use try...catch.

File: lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp [hunk 6]
```diff
@@ -187,8 +186,7 @@ sgw_create_bearer_context_information_in_collection(
 
   *temporary_create_session_procedure_id_p = (uint32_t)rand();
   sgw_eps_bearer_context_information_t* new_sgw_bearer_context_information =
-      reinterpret_cast<sgw_eps_bearer_context_information_t*>(
-          calloc(1, sizeof(sgw_eps_bearer_context_information_t)));
+      new sgw_eps_bearer_context_information_t();
 
   if (new_sgw_bearer_context_information == NULL) {
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp]: nullptr check does not work here. Remove it or use try..catch.

File: lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp [hunk 7]
```diff
@@ -95,9 +95,9 @@ status_code_e sgw_handle_s11_create_session_request(
     spgw_state_t* state,
     const itti_s11_create_session_request_t* const session_req_pP,
     imsi64_t imsi64) {
-  mme_sgw_tunnel_t* new_endpoint_p = NULL;
+  mme_sgw_tunnel_t* new_endpoint_p = nullptr;
   s_plus_p_gw_eps_bearer_context_information_t*
-      s_plus_p_gw_eps_bearer_ctxt_info_p = NULL;
+      s_plus_p_gw_eps_bearer_ctxt_info_p = nullptr;
   sgw_eps_bearer_ctxt_t* eps_bearer_ctxt_p = NULL;
```
Comment (ajahl) [inline lte/gateway/c/core/oai/tasks/sgw/sgw_handlers.cpp]: sgw_eps_bearer_ctxt_t* eps_bearer_ctxt_p = nullptr;

**Comment by pruthvihebbani** [comment]
@rsarwad Can you please rebase to resolve insync checkin failure

**Review by pruthvihebbani — APPROVED** [review]
LGTM

**Review by ajahl — APPROVED** [review]

**Pull request merged**